straxen/plugins/peak_processing.py

```diff
@@ -332,97 +334,316 @@ def find_n_competing(peaks, windows, fraction):
+        num_array = np.zeros(len(current_peak), dtype=np.int16)
+        sum_array = np.zeros(len(current_peak), dtype=np.float32)
+        mask_pre = (peaks['type'] == 2) & (peaks['area'] < self.ambience_area_parameters[2])
+        touching_windows = strax.touching_windows(peaks[mask_pre], roi)
+        # Calculating ambience
+        self.peaks_ambience(current_peak, peaks[mask_pre], touching_windows, radius, num_array, sum_array, self.ambience_divide_time, self.ambience_divide_space)
+        result['n_s2_near'] = num_array
+        result['s_s2_near'] = sum_array
+
+        # 6. Set time and endtime for peaks
+        result['time'] = current_peak['time']
+        result['endtime'] = strax.endtime(current_peak)
+        return result
+
+    @staticmethod
+    @numba.njit
+    def lonehits_ambience(peaks, pre_hits, touching_windows, num_array, sum_array, ambience_divide_time):
+        # Function to find lonehits before a peak
+        for p_i, p_a in enumerate(peaks):
+            indices = touching_windows[p_i]
+            for idx in range(indices[0], indices[1]):
+                s_a = pre_hits[idx]
+                dt = p_a['center_time'] - s_a['time']
+                if dt <= 0:
+                    continue
+                num_array[p_i] += 1
+                # Sometimes we may interested in sum of area / dt
+                if ambience_divide_time:
+                    sum_array[p_i] += s_a['area'] / dt
+                else:
+                    sum_array[p_i] += s_a['area']
+
+    @staticmethod
+    @numba.njit
+    def peaks_ambience(peaks, pre_peaks, touching_windows, radius, num_array, sum_array, ambience_divide_time, ambience_divide_space):
+        # Function to find S0, S1, S2 before or near a peak
         for p_i, p_a in enumerate(peaks):
-            # reset for every peak
-            new_shadow = 0
-            s2_indices = touching_windows[p_i]
-            for s2_idx in range(s2_indices[0], s2_indices[1]):
-                s2_a = pre_s2_peaks[s2_idx]
-                if p_a['center_time'] - s2_a['center_time'] <= 0:
+            indices = touching_windows[p_i]
+            for idx in range(indices[0], indices[1]):
+                s_a = pre_peaks[idx]
+                r = ((p_a['x'] - s_a['x'])**2 + (p_a['y'] - s_a['y'])**2)**0.5
+                dt = p_a['center_time'] - s_a['center_time']
+                if dt <= 0:
                     continue
-                new_shadow = s2_a['area'] * (
-                        p_a['center_time'] - s2_a['center_time'])**exponent
-                if new_shadow > res['shadow'][p_i]:
-                    res['shadow'][p_i] = new_shadow
-                    res['pre_s2_area'][p_i] = s2_a['area']
-                    res['shadow_dt'][p_i] = p_a['center_time'] - s2_a['center_time']
-                    res['pre_s2_x'][p_i] = s2_a['x']
-                    res['pre_s2_y'][p_i] = s2_a['y']
+                if (dt != 0) and ((radius < 0) or (r <= radius)):
```

Comment:
`radius` is a bit general. Maybe call it max_ambient_radius?

Reply:
You can drop the test for `dt` here. Given the `dt <= 0` above you know that all dts must be positive here.